### PR TITLE
Add resource requirements to sidecar agent

### PR DESCRIPTION
**Pull request by objectiser**
Related to #169 

Adds resource requirements to the agent sidecar.

Signed-off-by: Gary Brown <gary@brownuk.com>

### CODE REVIEW

**Comment by objectiser** [comment]
It doesn't answer the issue raised in #70 related to how to define resource requirements if using both daemonset and sidecar agents, but addresses the more immediate issue of not being able to define resource limits for sidecar agents, when only that approach is being used.

So currently the resource limits defined against the agent will be used for both daemonset and sidecar agents.

**Comment by pavolloffay** [comment]
Shouldn't we have more fine-grained control over agent sidecar resources? There is already a PR which sets the resources based on annotations https://github.com/jaegertracing/jaeger-operator/pull/169. 

I think this could go in as default global setting and be able to override it via annotations.

File: pkg/inject/sidecar.go [hunk 1]
```diff
@@ -104,6 +104,8 @@ func container(jaeger *v1.Jaeger) corev1.Container {
 	jgCompactTrft := util.GetPort("--processor.jaeger-compact.server-host-port=", args, 6831)
 	jgBinaryTrft := util.GetPort("--processor.jaeger-binary.server-host-port=", args, 6832)
 
+	commonSpec := util.Merge([]v1.JaegerCommonSpec{jaeger.Spec.Agent.JaegerCommonSpec, jaeger.Spec.JaegerCommonSpec})
```
Comment (pavolloffay) [inline pkg/inject/sidecar.go]: What does this do? Why isn't it enough to use just `Spec.Agent.Resources`?

Comment (objectiser) [reply pkg/inject/sidecar.go]: It allows common resource requirements to be defined in `jaeger.Spec.Resources` and only agent specific requirements defined in `jaeger.Spec.Agent.Resources`. Its a pattern used for all parts of the common spec (e.g. volumnes, mounts, annotations, etc).

**Comment by objectiser** [comment]
@pavolloffay I've removed the `closes 169`, so that will now remain open, so that it can be added on top of this PR to provide annotation based approach aswell.

File: pkg/inject/sidecar.go (hunk 1 above)
Comment (objectiser) [reply pkg/inject/sidecar.go]: Yes. It has been done from a consistency perspective - but if is not reasonable, then we just need to make sure it is documented clearly that for agent as a sidecar (or agent in general), the parent resources are not inherited.

Comment (pavolloffay) [reply pkg/inject/sidecar.go]: I would maybe prefer applying only resources on the agent spec as sidecar can populate to a lot of pods but I don't have strong objections.

**Comment by objectiser** [comment]
@pavolloffay I'll keep like this for now, just for consistency. Most people will probably just set the requirements on the Agent itself - so if we decide to change later it probably won't have a big impact.

@jpkrohling When you get a chance, would be good to get your thoughts on this.

**Pull request merged**